### PR TITLE
feature/deseng750: Added compatibility for multiple shapefiles.

### PR DESCRIPTION
**Pull request by jareth-whitney**
- Added compatibility for multiple shape files
- Fixed long-standing issue with map going blank when navigating between two map pages

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
-### 1.15 Nov 27, 2024
+### 1.15 Dec 9, 2024
 
 - Agreements header is now invisible when no agreements are available. [DESENG-742](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-742)
 - Shapefiles can now appear with custom colours on the main project map and the individual project map. [DESENG-743](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-743)
 - The Project Phase column in the project list has been replaced with Process Type(s). It lists relevant project planning types. [DESENG-746](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-746)
 - Custom collection notice will appear in contact and subscribe modals when available.  [DESENG-747](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-747)
 - Updated "Learn More" routes to reflect their current content. [DESENG-748](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-748)
+- Added full compatibility for multiple shape files in a single project, using one or multiple zip files. [DESENG-750](https://apps.itsm.gov.bc.ca/jira/browse/DESENG-750)
```
Comment (Baelx) [inline CHANGELOG.md]: Can you clarify if the rest of these features have been pushed to dev/prod? If they have, you'd wanna create a new date heading for this latest feature

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 2]
```diff
@@ -46,12 +51,12 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
     this.overlappingDistrictsListString = this.stringifyOverlappingDistricts(this.project.overlappingRegionalDistricts);
     this.commentPeriod = this.project.commentPeriodForBanner;
     this.route.data.subscribe((res: any) => {
-      if (res) {
-        if (res.documents && res.documents[0].data.meta && res.documents[0].data.meta.length > 0) {
-          this.shapefile = res.documents[0].data.searchResults;
-        } else {
-          this.shapefile = [];
-        }
+      if (Array.isArray(res?.documents)) {
```
Comment (Baelx) [inline src/app/project/project-details-tab/project-details-tab.component.ts:54]: Thanks for making this code a bit cleaner!

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 3]
```diff
@@ -156,40 +161,112 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
       this.configService.baseLayerName = e.name;
     }, this);
 
-    // Disable zoom on project details - iterferes with scrolling page.
+    // Disable mouse zoom on project details - iterferes with scrolling page.
```
Comment (Baelx) [inline src/app/project/project-details-tab/project-details-tab.component.ts:164]: Appreciate you clarifying here!

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 4]
```diff
@@ -156,40 +161,112 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
...
+			if (this.project) {
+				this.addMarker();
+				this.map.addLayer(this.appFG);
+				this.map.fitBounds(this.defaultBounds, {padding: [50, 50]});
+			}
+		}
+	}
+
+	/**
+	 * Caluclates the shapefile bounds.
```
Comment (Baelx) [inline src/app/project/project-details-tab/project-details-tab.component.ts]:
```suggestion
	 * Calculates the shapefile bounds.
```

Also wouldn't it be more accurate to say: "Takes the bounds of a shapefile and updates the overall map bounds to include them."

**Review by Baelx — COMMENTED** [review]
Thanks for fixing an old bug and adding some cleaning up some of the code/comments.

Could I just ask you to clarify if/how the code will be deployed? That will affect how we update the Changelog. If you've already pushed all the other recent changes to dev, then you'll need a separate date heading for this PR's changes.

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 4 above)
Comment (jareth-whitney) [reply src/app/project/project-details-tab/project-details-tab.component.ts]: Sure, I switched the comment to this.

File: CHANGELOG.md (hunk 1 above)
Comment (jareth-whitney) [reply CHANGELOG.md]: I have updated to version 1.16 and updated the change log.

**Review by Baelx — APPROVED** [review]

**Pull request merged**